packages/core/src/interpreter.ts

```diff
@@ -519,6 +519,12 @@ export class Interpreter<
       return this;
     }
 
+    // yes, it's a hack but we need the related cache to be populated for some things to work (like delayed transitions)
+    // this is usually called by `machine.getInitialState` but if we rehydrate from a state we might bypass this call
+    // we also don't want to call this method here as it resolves the full initial state which might involve calling assign actions
+    // and that could potentially lead to some unwanted side-effects (even such as creating some rogue actors)
+    (this.machine as any)._init();
```

Comment:
This should really only be called here: https://github.com/statelyai/xstate/blob/c3e2b75ea02622fbc788782ef96229ea90046d11/packages/core/src/StateNode.ts#L464-L468

but we are still blocked by supporting "getters" for targets so we can't call this in a constructor as the `machine` doesn't get yet returned to the caller at this point in time and thus a thing like this fails on reading from `undefined`:
https://github.com/statelyai/xstate/blob/c3e2b75ea02622fbc788782ef96229ea90046d11/packages/core/test/id.test.ts#L52-L54